src/Compiler/Service/TransparentCompiler.fs

```diff
@@ -670,11 +673,27 @@ type internal TransparentCompiler
             return tcInfo, sink, implFile, fileName
         }
 
-    let mergeTcInfos =
-        Array.fold (fun a b ->
+    let mergeIntermediateResults bootstrapInfo =
+        Array.reduce (fun (a: TcInfo, _: TcResultsSinkImpl, _, _) (b, bSink, implFileOpt: CheckedImplFile option, name) ->
             // TODO: proper merge
+
+            let amap = bootstrapInfo.TcImports.GetImportMap()
+
+            // TODO: figure out
+            let hadSig = false
+
+            let prefixPathOpt = None
+
+            // TODO:
+            let implFile = implFileOpt.Value
+
+            let _ccuSigForFile, tcState =
+                AddCheckResultsToTcState
```

Comment:
If two files share a dependency upstream, do we need to avoid adding it twice to the state?

Reply:
Possibly. This is still highly experimental 😄 

Though now the simplest tests are passing without having to deal with that.

Reply:
This is now handled by the callback system.